src/ChunkSplitters.jl

```diff
@@ -56,28 +71,85 @@ end
 
 import Base: length, eltype
 length(c::Chunk) = c.nchunks
-eltype(::Chunk) = Tuple{StepRange{Int,Int},Int}
+eltype(::Chunk) = StepRange{Int,Int}
 
 import Base: firstindex, lastindex, getindex
 firstindex(::Chunk) = 1
 lastindex(c::Chunk) = c.nchunks
-getindex(c::Chunk, i::Int) = (getchunk(c.x, i, c.nchunks, c.type), i)
-
-import Base: collect
-collect(c::Chunk) = [(getchunk(c.x, i, c.nchunks, c.type), i) for i in 1:c.nchunks]
+getindex(c::Chunk, i::Int) = getchunk(c.x, i, c.nchunks, c.type)
 
 #
 # Iteration of the chunks
 #
 import Base: iterate
 function iterate(c::Chunk, state=nothing)
     if isnothing(state)
-        return ((getchunk(c.x, 1, c.nchunks, c.type), 1), 1)
+        chunk = getchunk(c.x, 1, c.nchunks, c.type)
+        return (chunk, 1)
     elseif state < c.nchunks
-        return ((getchunk(c.x, state + 1, c.nchunks, c.type), state + 1), state + 1)
-    else
-        return nothing
+        chunk = getchunk(c.x, state + 1, c.nchunks, c.type)
+        return (chunk, state + 1)
     end
+    return nothing
+end
+
+#
+# Iteration over chunks enumeration: usually enumerate is not compatible
+# with `@threads`, because of the lack of the general definition of 
+# `firstindex`, `lastindex`, and `getindex` for `Base.Iterators.Enumerate{<:Any}`. Thus,
+# to avoid using the internal `.itr` property of `Enumerate`, we redefine
+# the `iterate` method for `ChunkSplitters.Enumerate{<:Chunk}`.
+#
+struct Enumerate{I<:Chunk}
+    itr::I
+end
+import Base: enumerate
+enumerate(c::Chunk) = Enumerate(c)
+
+import Base: iterate
+function iterate(ec::Enumerate{<:Chunk}, state=nothing)
```

Comment:
```suggestion
function Base.iterate(ec::Enumerate{<:Chunk}, state=nothing)
```

Reply:
I recommend doing the same for the rest, i.e. dropping the `import` and putting `Base.` in front of the definition. This makes it very clear that a Base function is overloaded here.